go/filemanager/pfsmodules/chunkmeta.go

```diff
@@ -0,0 +1,218 @@
+package pfsmodules
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"strconv"
+)
+
+const (
+	defaultMaxChunkSize = 4 * 1024 * 1024
+	defaultMinChunkSize = 4 * 1024
+)
+const (
+	chunkMetaCmdName = "GetChunkMeta"
+)
+
+// ChunkMeta holds the chunk meta's info
+type ChunkMeta struct {
+	Offset   int64  `json:"offset"`
+	Checksum string `json:"checksum"`
+	Len      int64  `json:"len"`
+}
+
+// ChunkMetaCmd is a command
+type ChunkMetaCmd struct {
+	Method    string `json:"method"`
+	FilePath  string `json:"path"`
+	ChunkSize int64  `json:"chunksize"`
+}
+
+// ToURLParam encodes ChunkMetaCmd to URL encoding string
+func (p *ChunkMetaCmd) ToURLParam() string {
+	parameters := url.Values{}
+	parameters.Add("method", p.Method)
+	parameters.Add("path", p.FilePath)
+
+	str := fmt.Sprint(p.ChunkSize)
+	parameters.Add("chunksize", str)
+
+	return parameters.Encode()
+}
+
+// ToJSON encodes ChunkMetaCmd to JSON string
+func (p *ChunkMetaCmd) ToJSON() ([]byte, error) {
+	return json.Marshal(p)
+}
+
+// Run is a functions which run ChunkMetaCmd
+func (p *ChunkMetaCmd) Run() (interface{}, error) {
+	return GetChunkMeta(p.FilePath, p.ChunkSize)
+}
+
+func (p *ChunkMetaCmd) checkChunkSize() error {
+	if p.ChunkSize < defaultMinChunkSize ||
+		p.ChunkSize > defaultMaxChunkSize {
+		return errors.New(StatusBadChunkSize)
+	}
+
+	return nil
+}
+
+// CloudCheck checks the conditions when running on cloud
+func (p *ChunkMetaCmd) CloudCheck() error {
+	if !IsCloudPath(p.FilePath) {
+		return errors.New(StatusShouldBePfsPath + ": p.FilePath")
+	}
+	if !CheckUser(p.FilePath) {
+		return errors.New(StatusUnAuthorized + ":" + p.FilePath)
+	}
+	return p.checkChunkSize()
+}
+
+// LocalCheck checks the conditions when running local
+func (p *ChunkMetaCmd) LocalCheck() error {
+	return p.checkChunkSize()
+}
+
+// NewChunkMetaCmdFromURLParam get a new ChunkMetaCmd
+func NewChunkMetaCmdFromURLParam(r *http.Request) (*ChunkMetaCmd, error) {
+	method := r.URL.Query().Get("method")
+	path := r.URL.Query().Get("path")
+	chunkStr := r.URL.Query().Get("chunksize")
+
+	if len(method) == 0 ||
+		method != chunkMetaCmdName ||
+		len(path) < 4 ||
+		len(chunkStr) == 0 {
+		return nil, errors.New(http.StatusText(http.StatusBadRequest))
+	}
+
+	chunkSize, err := strconv.ParseInt(chunkStr, 10, 64)
+	if err != nil {
+		return nil, errors.New(StatusBadChunkSize)
+	}
+
+	return &ChunkMetaCmd{
```

Comment:
Here is using struct literal (`&ChunkMetaCmd{...}`), but there is a function for it below `NewChunkMetaCmd` which does exactly the same thing. I suggest removing function `NewChunkMetaCmd` and `NewChunk`, and just use the struct literal way.

To reduce complexity, I recommend unless there is some other initialization that a struct literal can not do, don't use a new function for it.